collect_app/src/main/java/org/odk/collect/android/preferences/QRCodeTabs.java

```diff
@@ -0,0 +1,128 @@
+package org.odk.collect.android.preferences;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.odk.collect.android.BuildConfig;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.CollectAbstractActivity;
+import org.odk.collect.android.activities.MainMenuActivity;
+import org.odk.collect.android.adapters.TabAdapter;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.listeners.ActionListener;
+import org.odk.collect.android.utilities.FileUtils;
+import org.odk.collect.android.utilities.LocaleHelper;
+import org.odk.collect.android.utilities.QRCodeUtils;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import java.io.File;
+
+import timber.log.Timber;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.FileProvider;
+import androidx.viewpager2.widget.ViewPager2;
+
+import com.google.android.material.tabs.TabLayout;
+import com.google.android.material.tabs.TabLayoutMediator;
+
+import static org.odk.collect.android.activities.ActivityUtils.startActivityAndCloseAllOthers;
+
+public class QRCodeTabs extends CollectAbstractActivity {
```

Comment:
done! @seadowg should this also be moved to collect/android/activities folder?

Reply:
As above I think it moves to `preferences`. Maybe all of it should go in a `preferences.qr` packakge... what do you think? We're trying to organize our code by function/domain rather than component type.

Reply:
I like the idea of a `preferences.qr` package. There's a few files in here and it feels a lot cleaner than forcing even more items into the crowded `preferences` package.